src/libp2p/Libp2p.Protocols.Noise/NoiseProtocol.cs

```diff
@@ -60,6 +60,37 @@ public async Task DialAsync(IChannel downChannel, IChannelFactory? upChannelFact
         NoiseHandshakePayload? msg1Decoded = NoiseHandshakePayload.Parser.ParseFrom(buffer.AsSpan(0, msg1.BytesRead));
         PublicKey? msg1KeyDecoded = PublicKey.Parser.ParseFrom(msg1Decoded.IdentityKey);
         //var key = new byte[] { 0x1 }.Concat(clientStatic.PublicKey).ToArray();
+        List<string> responderMuxers = msg1Decoded.Extensions.StreamMuxers
+    .Where(m => !string.IsNullOrEmpty(m))
+    .ToList();
+
+        IProtocol commonMuxer = null;
+
+        foreach (var responderMuxer in responderMuxers)
```

Comment:
Can be expressed in smaller form: `IProtocol commonMuxer = multiplexerSettings.Multiplexers.FirstOrDefault(m=>responderMuxers.Contains(m.Id));`
and below can be `IProtocol commonMuxer = responderMuxers.FirstOrDefault(rm=>multiplexerSettings.Multiplexers.Any(m => m.Id == rm));`
to respect initiator choice of ordering multiplexers.
(Sounds complicated, let's have a call if needed)